### PR TITLE
Feature/vtn 28641 migrate react component

### PR DESCRIPTION
**Pull request by seta-tuha**
Ticket: https://steel-ventures.atlassian.net/browse/VTN-28641
Purpose: Migrate react to version 16.9.0 and material-ui to 4.4

### CODE REVIEW

File: packages/veritone-react-common/src/components/DataPicker/UploaderViewContainer/index.js [hunk 1]
```diff
@@ -46,75 +45,77 @@ const UploadViewContainer = ({
...
-      && percentByFiles.length > 0
-      && percentByFiles.some(percent => percent.value.percent > 0)
-    ) || (uploadedFiles && !uploadedFiles.length);
+    percentByFiles
+    && percentByFiles.length > 0
+    && percentByFiles.some(percent => percent.value.percent > 0)
+  ) || (uploadedFiles && !uploadedFiles.length);
 
   return (
     <div className={styles['uploader-container']} style={containerStyle}>
```
Comment (MmtBkn) [inline packages/veritone-react-common/src/components/DataPicker/UploaderViewContainer/index.js]: @SteveShaffer Shall we use kabob case or camel case for classes (just asking for consistency)?

File: packages/veritone-react-common/src/helpers/withVeritoneSDKThemeProvider.js [hunk 2]
```diff
@@ -27,27 +28,32 @@ export const defaultVSDKTheme = {
   }
 };
 
+const jss = create({
+  ...jssPreset()
+});
+
+const generateClassName = createGenerateClassName({
```
Comment (MmtBkn) [inline packages/veritone-react-common/src/helpers/withVeritoneSDKThemeProvider.js]: This should be passed by client, If client doesn't pass one, we can create it.

https://material-ui.com/getting-started/faq/#why-arent-my-components-rendering-correctly-in-production-builds

File: packages/veritone-react-common/package.json [hunk 3]
```diff
@@ -5,23 +5,26 @@
...
-    "@material-ui/icons": "^2.0.0",
+    "@material-ui/core": "^4.3.3",
+    "@material-ui/icons": "^4.2.1",
+    "@material-ui/lab": "^4.0.0-alpha.24",
+    "@material-ui/styles": "^4.3.3",
     "classnames": "^2.2.5",
     "clipboard": "^2.0.4",
     "date-fns": "^1.29.0",
     "google-maps": "^3.3.0",
+    "jss": "10.0.0-alpha.17",
```
Comment (MmtBkn) [inline packages/veritone-react-common/package.json]: @SteveShaffer are we ok with alpha version here?

Comment (seta-tuha) [reply packages/veritone-react-common/package.json]: I am following this to fix the problem with jss: https://github.com/mui-org/material-ui/issues/12362

File: packages/veritone-react-common/src/components/DataPicker/UploaderViewContainer/index.js (hunk 1 above)
Comment (seta-tuha) [reply packages/veritone-react-common/src/components/DataPicker/UploaderViewContainer/index.js]: @MmtBkn, @SteveShaffer  In my opinion, we should use kabob case for classes in .css files, but for .scss module I am not sure

File: packages/veritone-react-common/src/helpers/withVeritoneSDKThemeProvider.js (hunk 2 above)
Comment (seta-tuha) [reply packages/veritone-react-common/src/helpers/withVeritoneSDKThemeProvider.js]: Done! passing via props

**Comment by seta-tuha** [comment]
@SteveShaffer @twoo24 @MmtBkn Please help me to review this PR quickly as other PRs are currently depending on this. Thanks

File: packages/veritone-react-common/package.json [hunk 4]
```diff
@@ -123,12 +125,18 @@
       "^helpers(.*)$": "<rootDir>/src/helpers$1",
       "^components(.*)$": "<rootDir>/src/components$1",
       "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/test/__mocks__/fileMock.js",
-      "\\.(css|scss)$": "identity-obj-proxy"
+      "\\.(css|scss)$": "identity-obj-proxy",
+      "^dnd-core$": "dnd-core/dist/cjs",
```
Comment (MmtBkn) [inline packages/veritone-react-common/package.json]: @seta-tuha

Comment (seta-tuha) [reply packages/veritone-react-common/package.json]: It's just configuration for dnd test

File: packages/veritone-react-common/src/components/DataPicker/UploaderViewContainer/index.js (hunk 1 above)
Comment (SteveShaffer) [reply packages/veritone-react-common/src/components/DataPicker/UploaderViewContainer/index.js]: It looks like camelCase is the consistent option.  If the classes were gonna hit the DOM I would say kebab-case, but with classNames, those are just reference names anyway so please go with camelCase.

File: packages/veritone-react-common/package.json (hunk 3 above)
Comment (SteveShaffer) [reply packages/veritone-react-common/package.json]: Yes I'm fine with this case.

btw something to note: our mix of bundled-dependencies-that-all-rely-on-eachother includes React, Redux, Material-UI, and now JSS (and I presume react-jss?)

File: packages/veritone-react-common/package.json (hunk 4 above)
Comment (SteveShaffer) [reply packages/veritone-react-common/package.json]: I'm fine with this instance since it's just in jest

**Comment by SteveShaffer** [comment]
Wait @seta-tuha you have 1 failing test.  Please resolve before merging.
```
 |   282 passing (4s)
 |   7 pending
 |   1 failing
 | 
 |   1) callApi (Node) should ultimately fail after exhausting retries (promise):
 |      Uncaught TypeError: Cannot read property 'worked' of null
 |       at packages/veritone-client-js/apis/helper/callApi.spec.js:675:25
 |       at <anonymous>
 |       at process._tickCallback packages/veritone-client-js/(internal/process/next_tick.js:189:7)
```

**Comment by seta-tuha** [comment]
> I've resolved all these conversations. Thanks for your work on this @seta-tuha. Let's make sure we're uber attentive to UI breakages when upgrading components after this. It's a big upgrade!



> Wait @seta-tuha you have 1 failing test. Please resolve before merging.
> 
> ```
>  |   282 passing (4s)
>  |   7 pending
>  |   1 failing
>  | 
>  |   1) callApi (Node) should ultimately fail after exhausting retries (promise):
>  |      Uncaught TypeError: Cannot read property 'worked' of null
>  |       at packages/veritone-client-js/apis/helper/callApi.spec.js:675:25
>  |       at <anonymous>
>  |       at process._tickCallback packages/veritone-client-js/(internal/process/next_tick.js:189:7)
> ```

Actually this error comes from 'veritone-client-js', I run test at local does not occur any error :|

**Pull request merged**